### PR TITLE
Refactor the code

### PR DESCRIPTION
**Pull request by RidmiR**
## Purpose
Refactor the code of MicroRegistryManager. #544

### CODE REVIEW

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java [hunk 1]
```diff
@@ -123,19 +123,26 @@ public void updateResource(String resourcePath, String sourcePath, boolean backu
...
-     * @param sourcePath path to registry resource
      * @param resourcePath path to resource file path from registry resource
      * @param filename registry filename
      */
 
-    public boolean checkResourceExist(String sourcePath, String resourcePath, String filename) throws Exception {
-        try {
-            String path = sourcePath + resourcePath + filename;
-            return path.contains(filename);
+    public boolean checkResourceExist(String resourcePath, String filename) throws Exception {
```
Comment (sdkottegoda) [inline integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: Do we need to throw a generic exception?

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java [hunk 2]
```diff
@@ -123,19 +123,26 @@ public void updateResource(String resourcePath, String sourcePath, boolean backu
...
+        pathBuilder.append(System.getProperty("carbon.home")).append(File.separator).append("registry").append(File.separator);
 
-        } catch (Exception e) {
-            throw new Exception("Exception occurred while checking the registry resource file.", e);
+        if (resourcePath.startsWith(CONF_PATH)) {
+            String relativePath = resourcePath.substring(5).replace('/', File.separatorChar);
+            pathBuilder.append("config").append(relativePath);
+        } else if (resourcePath.startsWith(GOV_PATH)) {
+            String relativePath = resourcePath.substring(4).replace('/', File.separatorChar);
+            pathBuilder.append("governance").append(relativePath);
```
Comment (sdkottegoda) [inline integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: We can extract the duplicated code into a separate method. Shall we?

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java [hunk 3]
```diff
@@ -123,19 +123,26 @@ public void updateResource(String resourcePath, String sourcePath, boolean backu
...
+        } else {
+            throw new MicroRegistryManagerException("Exception occurred while checking the registry resource file: " + resourcePath);
```
Comment (sdkottegoda) [inline integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: Would it be more meaningful if we say invalid resource path + path. The resource path should either start with conf: or gov: ?

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java (hunk 2 above)
Comment (RidmiR) [reply integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: Added a new method to gov and conf. Please check the latest commit.

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java (hunk 3 above)
Comment (RidmiR) [reply integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: Changed the message. Please review in the latest commit.

File: integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java (hunk 1 above)
Comment (RidmiR) [reply integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/MicroRegistryManager.java]: Done

**Pull request closed without merging**